lib/puppet-lint/report/codeclimate.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'digest'
+require 'json'
+
+class PuppetLint::Report
+  class CodeClimateReporter
+    def self.write_report_file(problems, report_file)
+      report = []
+      problems.each do |messages|
+        messages.each do |message|
+          case message[:kind].downcase
+          when :warning
+            severity = 'minor'
+          when :error
+            severity = 'major'
+          else
+            next
+          end
+
+          issue = {
+            type: :issue,
+            check_name: message[:check],
+            description: message[:message],
+            categories: [:Style],
+            severity: severity,
+            location: {
+              path: message[:path],
+              lines: {
+                begin: message[:line],
+                end: message[:line],
+              }
+            },
+            fingerprint: Digest::MD5.hexdigest(Marshal.dump(message))
+          }
+
+          if message.key?(:description) && message.key?(:help_uri)
+            issue[:content] = "#{message[:description].chomp('.')}. See [this page](#{message[:help_uri]}) for more information about the `#{message[:check]}` check."
+          end
+          report << issue
+        end
+      end
+      File.open(report_file, 'w') { |f| f.write JSON.pretty_generate(report) }
```

Comment:
Or just write it at once.
```suggestion
      File.write(report_file, JSON.pretty_generate(report))
```